src/Components/Nav/Nav.jsx

```diff
@@ -1,41 +1,140 @@
+            </span>
+            {searchValue !== "" && (
+              <span
+                className="cross-icon"
+                onClick={() => {
+                  setSearchValue("");
+                  dispatch({ type: "SEARCHRESET" });
+                }}
+              >
+                <i class="fas fa-times"></i>
+              </span>
+            )}
+            <div>
+              <div className="filter-container">
+                <p
+                  className="filter-title"
+                  onClick={() => setAppearFilter((prev) => !prev)}
+                >
+                  Sort by date <i class="fas fa-calendar-alt"></i>
+                </p>
+                <p
+                  className="clear-filter"
+                  onClick={() => {
+                    setSearchValue("");
+                    dispatch({ type: "CLEAR" });
+                  }}
+                >
+                  Clear all filters
+                </p>
+                {appearFilter && (
+                  <div className="radio-filters">
+                    <div className="filter-element">
+                      <input
+                        value="newestFirst"
+                        type="radio"
+                        id="new"
+                        name="radio-filter"
+                        onClick={(e) =>
+                          dispatch({
+                            type: "DATEFILTER",
+                            payload: e.target.value,
+                          })
+                        }
+                      />
+                      <label className="filter-label" htmlFor="new">
+                        Newest First
+                      </label>
+                    </div>
+                    <div className="filter-element">
+                      <input
+                        value="oldestFirst"
+                        type="radio"
+                        id="old"
+                        name="radio-filter"
+                        onClick={(e) =>
+                          dispatch({
+                            type: "DATEFILTER",
+                            payload: e.target.value,
+                          })
+                        }
+                      />
+                      <label className="filter-label" htmlFor="old">
+                        Oldest First
+                      </label>
+                    </div>
+                  </div>
+                )}
+              </div>
+            </div>
           </div>
 
-          <i className="fab fa-github icon"></i>
-          <i className="fab fa-twitter icon"></i>
+          <div className="icons">
+            <div className="user-icon-container">
+              <p
+                className="logout-text"
+                style={{ display: appear ? "block" : "none" }}
+                onClick={() => {
+                  logoutHandler();
+                  navigate("/");
+                  setAppear((prev) => !prev);
```

Comment:
this might throw some error or warning. Always perform state update before routing
Post routing, react will try to set a state of an unmounted component, which might create problems.